shared/metadata.go

```diff
@@ -20,3 +54,81 @@ type MetadataLink struct {
 	TestPath string `yaml:"test"`
 	URL      string
 }
+
+func (m MetadataResults) Len() int           { return len(m) }
+func (m MetadataResults) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
+func (m MetadataResults) Less(i, j int) bool { return m[i].Test < m[j].Test }
+
+// GetMetadataResponse retrieves the response to a WPT Metadata query.
+func GetMetadataResponse(testRuns []TestRun, client *http.Client) MetadataResponse {
+	metadataByteMap, err := util.CollectMetadata(client)
+	if err != nil {
+		return MetadataResponse{}
+	}
+	metadata := parseMetadata(metadataByteMap)
+	return constructMetadataResponse(testRuns, metadata)
+}
+
+// parseMetadata collects and parses all META.yml files from
+// wpt-metadata reposiroty.
+func parseMetadata(metadataByteMap map[string][]byte) map[string]Metadata {
+	var metadataMap = make(map[string]Metadata)
+	for path, data := range metadataByteMap {
+		var metadata Metadata
+		err := yaml.Unmarshal(data, &metadata)
+		if err != nil {
+			log.Warningf("Failed to unmarshal %s.yml.", path)
```

Comment:
This file is in the shared package. We usually try to make sure `shared` can run in both Standard and Flex environments. Different loggers are used in the two environments, so instead of using `logrus` directly, we should take a `shared.Logger` here and let the caller create the logger.